src/lib/storage/dictionary_segment.hpp

```diff
@@ -25,55 +30,141 @@ class DictionarySegment : public BaseSegment {
   /**
    * Creates a Dictionary segment from a given value segment.
    */
-  explicit DictionarySegment(const std::shared_ptr<BaseSegment>& base_segment);
+
+  explicit DictionarySegment(const std::shared_ptr<BaseSegment>& base_segment) {
+    auto value_segment = std::static_pointer_cast<ValueSegment<T>>(base_segment);
+    _dictionary = _initialize_dictionary(value_segment);
+    int minimal_bits = _get_minimal_number_of_bits_for_dictionary_size(_dictionary->size());
+    _attribute_vector = _create_attribute_vector_with_type(minimal_bits, value_segment, _dictionary);
+  }
 
   // SEMINAR INFORMATION: Since most of these methods depend on the template parameter, you will have to implement
   // the DictionarySegment in this file. Replace the method signatures with actual implementations.
 
   // return the value at a certain position. If you want to write efficient operators, back off!
-  AllTypeVariant operator[](const ChunkOffset chunk_offset) const override;
+  AllTypeVariant operator[](const ChunkOffset chunk_offset) const override { return get(chunk_offset); };
 
   // return the value at a certain position.
-  T get(const size_t chunk_offset) const;
+  T get(const size_t chunk_offset) const {
+    ValueID value_id = (ValueID)_attribute_vector->get(chunk_offset);
+    return value_by_value_id(value_id);
+  }
 
   // dictionary segments are immutable
-  void append(const AllTypeVariant& val) override;
+  void append(const AllTypeVariant& val) override {
+    throw std::runtime_error("Append failed: DictionarySegment is immutable.");
+  }
 
   // returns an underlying dictionary
-  std::shared_ptr<const std::vector<T>> dictionary() const;
+  std::shared_ptr<const std::vector<T>> dictionary() const { return _dictionary; }
 
   // returns an underlying data structure
-  std::shared_ptr<const BaseAttributeVector> attribute_vector() const;
+  std::shared_ptr<BaseAttributeVector> attribute_vector() const { return _attribute_vector; }
```

Comment:
is there a reason why you removed the const?

```suggestion
  std::shared_ptr<const BaseAttributeVector> attribute_vector() const { return _attribute_vector; }
```